### PR TITLE
Delegating likes

### PR DESCRIPTION
**Pull request by andorp**
Fixes: #698

### CODE REVIEW

File: src/Persistent/Pure.hs [hunk 1]
```diff
@@ -737,11 +737,13 @@ findComment :: CommentKey -> MQuery Comment
...
 
 instance FromProto IdeaLike where
-    fromProto () = IdeaLike
+    fromProto p m = IdeaLike m (_protoIdeaLikeDelegate p)
 
 -- FIXME: Assumption: the given @AUID Idea@ MUST be in the DB.
-addLikeToIdea :: AUID Idea -> AddDb IdeaLike
-addLikeToIdea iid = addDb' (mkIdeaVoteLikeKey iid) (dbIdeaMap . at iid . _Just . ideaLikes)
+addLikeToIdea :: AUID Idea -> User -> AddDb IdeaLike
+addLikeToIdea iid user =
```
Comment (fisx) [inline src/Persistent/Pure.hs]: is user the delegatee or the delegate?  this function could use a line of documentation.

Comment (fisx) [inline src/Persistent/Pure.hs]: also, if one of `delegate` or `delegatee` is a good name for something, then `user` probably isn't.

File: src/Action.hs [hunk 2]
```diff
@@ -531,33 +531,50 @@ moveIdeaToTopic ideaId moveIdea = do
...
 -- * Vote Handling
 
+delegatedOperationOnIdea
+    :: ActionM m
+    => (AUID User -> AUID Idea -> EQuery (Maybe (User, a)))
+    -> (User -> User -> m ())
+    -> AUID Idea
+    -> m ()
+delegatedOperationOnIdea getOperationResult operation ideaId = do
+    -- Like idea
```
Comment (fisx) [inline src/Action.hs]: what does this comment mean?

File: src/Action.hs [hunk 3]
```diff
@@ -531,33 +531,50 @@ moveIdeaToTopic ideaId moveIdea = do
 
 -- * Vote Handling
 
+delegatedOperationOnIdea
```
Comment (fisx) [inline src/Action.hs]: write a haddock comment?

**Comment by fisx** [comment]
LGTM otherwise.

File: src/Persistent/Pure.hs (hunk 1 above)
Comment (andorp) [inline src/Persistent/Pure.hs]: The user who does the like. It can be delegate and delegatee for this idea.

Comment (andorp) [inline src/Persistent/Pure.hs]: Oh I see, sorry.

File: src/Action.hs (hunk 3 above)
Comment (fisx) [inline src/Action.hs]: arguments:

1. get the delegate value (together with an operation result that is ignored).  rename to `getDelegate`.
1. run a new operation with delegate and delegatee on
1. this idea.

delegatee is current user.

Comment (fisx) [inline src/Action.hs]: correction: *delegate* (the one who casts the vote) is current user!

**Pull request merged**